include/arch/arc/arch.h

```diff
@@ -155,7 +155,6 @@ extern "C" {
 #endif /* CONFIG_USERSPACE */
 
 
-#ifdef CONFIG_USERSPACE
```

Comment:
FYI @andrewboie @ruuddw

Reply:
this is arc, perhaps ask @ruuddw ? no objections here

Reply:
@vonhust fyi

Reply:
ok for me - is there any test that uses mem partitions without userspace? Might be other dependencies/issues besides this #ifdef

Reply:
@ruuddw thanks for looking at it.

> is there any test that uses mem partitions without userspace? 

No, I do not think so. We expose k_mem_partition outside user-space, in this PR, to be able to start using mem partitions for other types of regions (no-cache, guards, etc.) but this will come in next PR.

Note: this will be done for ARM only, but the kernel API should be generic; and that is why we generralized how those kernel structures are used in all ARCHes with memory protection (ARC, ARM, x86)